packages/solana/lib/src/rpc/json_rpc_client.dart

```diff
@@ -15,50 +16,161 @@ class JsonRpcClient {
   final Duration _timeout;
   int _lastId = 1;
 
+  Future<List<Map<String, dynamic>>> bulkRequest(
+    String method,
+    List<List<dynamic>> params,
+  ) async {
+    final requests = params
+        .map(
+          (p) => _JsonRpcSingleRequest(
+            method: method,
+            params: p,
+            id: (_lastId++).toString(),
+          ),
+        )
+        .toList(growable: false);
+
+    final response = await _postRequest(_JsonRpcRequest.bulk(requests));
+    if (response is _JsonRpcArrayResponse) {
+      final elements = response.array;
+      return elements
+          .map((_JsonRpcObjectResponse item) => item.data)
+          .toList(growable: false);
+    } else {
+      throw FormatException('unexpected jsonrpc response type');
+    }
+  }
+
   /// Calls the [method] jsonrpc-2.0 method with [params] parameters
   Future<Map<String, dynamic>> request(
     String method, {
     List<dynamic>? params,
   }) async {
-    final request = <String, dynamic>{
-      'jsonrpc': '2.0',
-      'id': (_lastId++).toString(),
-      'method': method,
-    };
-    // If no parameters were specified, skip this field
-    if (params != null) request['params'] = params;
-    // Perform the POST request
-    final http.Response response = await http
-        .post(
-      Uri.parse(_url),
-      headers: <String, String>{
-        'Content-Type': 'application/json; charset=UTF-8',
-      },
-      body: json.encode(request),
-    )
-        .timeout(
-      _timeout,
-      onTimeout: () {
-        throw TimeoutException('Timeout while calling $method');
-      },
+    final request = _JsonRpcSingleRequest(
+      id: (_lastId++).toString(),
+      method: method,
+      params: params,
     );
 
+    final response = await _postRequest(request);
+    if (response is _JsonRpcObjectResponse) {
+      return response.data;
+    } else {
+      throw FormatException('unexpected jsonrpc response type');
+    }
+  }
+
+  static FutureOr<Response> _onTimeout() {
```

Comment:
I don't see the profit of extracting this method.

Reply:
BTW: it was just to combat the very ugly formatting that resulted otherwise